hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/protocol/commands/ReconcileContainerCommand.java

```diff
@@ -85,4 +87,22 @@ public String toString() {
         ": containerId=" + getContainerID() +
         ", peerNodes=" + peerDatanodes;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ReconcileContainerCommand that = (ReconcileContainerCommand) o;
+    return getContainerID() == that.getContainerID() &&
+        Objects.equals(peerDatanodes, that.peerDatanodes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getContainerID());
```

Comment:
Should it include hash of the peer datanodes as well?

Reply:
Yes, updated in the next commit. Default `hashCode` implementation from `AbstractSet` will compute it from all peer UUIDs which should be fine.